javatests/arcs/android/storage/service/DeferredStoreTest.kt

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2021 Google LLC.
+ *
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ *
+ * Code distributed by Google as part of this project is also subject to an additional IP rights
+ * grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+package arcs.android.storage.service
+
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import arcs.core.crdt.CrdtData
+import arcs.core.crdt.CrdtOperation
+import arcs.core.data.EntityType
+import arcs.core.data.FieldType
+import arcs.core.data.Schema
+import arcs.core.data.SchemaFields
+import arcs.core.data.SingletonType
+import arcs.core.storage.Driver
+import arcs.core.storage.DriverFactory
+import arcs.core.storage.StoreOptions
+import arcs.core.storage.WriteBack
+import arcs.core.storage.testutil.DummyStorageKey
+import arcs.core.storage.testutil.FakeDriverFactory
+import com.google.common.truth.Truth.assertThat
+import com.nhaarman.mockitokotlin2.mock
+import com.nhaarman.mockitokotlin2.verifyZeroInteractions
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.runBlockingTest
+import org.junit.Test
+import org.junit.runner.RunWith
+
+/**
+ * Tests ensure that [DeferredStore] lazily creates a store.
+ */
+@RunWith(AndroidJUnit4::class)
+@OptIn(ExperimentalCoroutinesApi::class)
+class DeferredStoreTest {
+
+  @Test
+  fun deferredStore_construction_doesNotCreateActiveStore() = runBlockingTest {
+    // We can confirm an active store is not created if these factories / providers are not invoked.
+    val mockDriverFactory = mock<DriverFactory>()
+    val mockWriteBack = mock<WriteBack>()
```

Comment:
The mocks that are shared between tests can be moved to be member variables, then you don't have to redefine them in every test.

Reply:
Not done? I meant move the definitions of `mockDriverFactory` and `writeBack` up to be member variables in this test class.